File: sme-creditmind-ai-backend/app/schemas/transaction.py
from pydantic import BaseModel


class MonthlyRevenueOut(BaseModel):
    month: str
    revenue: float
    transactionCount: int
    uniqueCustomers: int
    avgTicketSize: float


class TransactionUploadResponse(BaseModel):
    merchantId: str
    totalImported: int
    monthsCovered: int
    message: str

File: sme-creditmind-ai/src/types/transaction.ts
export type PaymentMethod = "CARD" | "EWALLET" | "QR_CODE" | "BANK_TRANSFER" | "CASH";

export interface Transaction {
  id: string;
  merchantId: string;
  amount: number;
  currency: string;
  paymentMethod: PaymentMethod;
  terminalId?: string;
  customerId?: string;
  category?: string;
  description?: string;
  transactionAt: string;
}

export interface MonthlyRevenue {
  month: string;
  revenue: number;
  transactionCount: number;
  uniqueCustomers: number;
  avgTicketSize: number;
}
